File: spring-webmvc/src/main/java/com/jiwoon/practicewebmvc/member/MemberService.java
package com.jiwoon.practicewebmvc.member;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Component;

//@Component
public class MemberService {

    private final MemberRepository memberRepository;

    @Autowired
    public MemberService(MemberRepository memberRepository) {
        this.memberRepository = memberRepository;
    }

    public void signup(Member member) {
        memberRepository.save(member);
    }

    public Member findMember(Long memberId) {
        return memberRepository.findById(memberId);
    }

    // 테스트 용도
    public MemberRepository getMemberRepository() {
        return memberRepository;
    }
}

File: spring-webflux-kotlin/src/main/kotlin/com/bcp0109/springwebfluxkotlin/domain/MemberService.kt
package com.bcp0109.springwebfluxkotlin.domain

import org.slf4j.Logger
import org.slf4j.LoggerFactory
import org.springframework.stereotype.Service
import reactor.core.publisher.Mono

@Service
class MemberService(
    private val memberRepository: MemoryMemberRepository
) {
    val log: Logger = LoggerFactory.getLogger(MemberService::class.java)

    fun signup(name: String, age: Int): Mono<Member> {
        val member = Member(name, age)
        return memberRepository.save(member)
    }

    fun findMemberById(memberId: Long): Mono<Member> {
        return memberRepository.findById(memberId)
    }
}
